IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometryResource/Functions/IfcGradient/Expression.txt

```diff
@@ -1,2 +1,37 @@
-(GradientCurve : IfcGradientCurve) : IfcLengthMeasure;
-  RETURN(1);
+(GradientCurve : IfcGradientCurve, Position : IfcCurveMeasureSelect) : IfcLengthMeasure;
+    
+  REPEAT i := 1 TO (SIZEOF(GradientCurve.Segments) - 1);
+    Segment : IfcCurveSegment := GradientCurve.Segments[i];
+    NextSegment : IfcCurveSegment := GradientCurve.Segments[i + 1];
+    Location : IfcCartesianPoint := NextSegment\IfcCurveSegment.Placement\IfcAxis2Placement2D.Location\IfcCartesianPoint;
+    SegmentPlacement : IfcAxis2Placement2D := Segment.Placement\IfcAxis2Placement2D;
+    SegDistAlong : IfcParameterValue := SegmentPlacement.Location\IfcCartesianPoint.Coordinates[1];
+	
+    IF (Location.Coordinates[1] > Position)
+      CASE TRUE OF
+	  
+        'IFCCIRCLE' IN TYPEOF(Segment.ParentCurve) :
+          BEGIN
+            RETURN SegmentPlacement.Location\IfcCartesianPoint.Coordinates[2] + Segment.ParentCurve.Radius * (COS(ACOS(SegmentPlacement.P[1][1]) / 2 - Segment.SegmentLength) - (Position - SegDistAlong));
+          END;
+		  
+        'IFCCLOTHOID' IN TYPEOF(Segment.ParentCurve) :
+          BEGIN			
+            RETURN SegmentPlacement.Location\IfcCartesianPoint.Coordinates[2] + 3 ** -1 * Segment.SegmentLength ** 3 + (3 * IfcFactorial(7)) ** -1 * Segment.SegmentLength ** 7 + (5 * IfcFactorial(11)) ** -1 * Segment.SegmentLength ** 11;
+          END;		
```

Comment:
All other equations take into account `(Position - SegDistAlong)`, but I cannot see this here. Is this intentional?